### PR TITLE
wip updates-on-operator-chart-configurations-

### PR DESCRIPTION
**Pull request by abby-cyber**
1. updated nebula-operator chart configuration files
2. add a new method to obtain certificates locally for the operator client through environment variables.

### CODE REVIEW

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md [hunk 1]
```diff
@@ -103,11 +120,45 @@ The following provides examples of the configuration file to enable mTLS between
...
-        # Zone names CANNOT be modified once set.
-        # It's suggested to set an odd number of zones.
-        # zone_list: az1,az2,az3 
-        validate_session_timestamp: "false"
-        licenseManagerURL: "192.168.8.xxx:9119"
+        config:
+          # Zone names CANNOT be modified once set.
+          # It's suggested to set an odd number of zones.
+          # zone_list: az1,az2,az3 
+          validate_session_timestamp: "false"
```
Comment (wenhaocs) [inline docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md:127]: Can we change it to true? With the current kernel release, it's true.

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md [hunk 2]
```diff
@@ -240,45 +319,11 @@ The following provides examples of the configuration file to enable mTLS between
...
-            emptyDir:
-              medium: Memory
-        volumeMounts:
-          - name: credentials
-            mountPath: /usr/local/nebula/certs
-        licenseManagerURL: "192.168.8.xx:9119"
+        # Zone names CANNOT be modified once set.
+        # It's suggested to set an odd number of zones.
+        # zone_list: az1,az2,az3 
+        validate_session_timestamp: "false"
```
Comment (wenhaocs) [inline docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md:325]: ditto

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md [hunk 3]
```diff
@@ -373,19 +382,6 @@ The following provides examples of the configuration file to enable mTLS between
...
-  graph:
-    config:
-      ca_client_path: certs/root.crt
-      ca_path: certs/root.crt
-      cert_path: certs/server.crt
-      key_path: certs/server.key
-      enable_graph_ssl: "true"
-```
-
 To enable mTLS between services (Graph, Meta, and Storage), add the following fields under the `spec.metad.config`, `spec.graphd.config`, and `spec.storaged.config` respectively in the cluster configuration file.
```
Comment (wenhaocs) [inline docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md:385]: Please state explicitly that the `sslCert` section is also required in this case.

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md [hunk 4]
```diff
@@ -493,34 +502,92 @@ Details about these fields are as follows:
...
+      - name: credentials
+        emptyDir:
+          medium: Memory
+
+    extraVolumeMounts:
+      - name: credentials
+        mountPath: /credentials
+  ```
+  
+- In the cluster configuration file, create a Secret to pass in the certificates. Example:
```
Comment (wenhaocs) [inline docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md:557]: This is another options. You may add `Alternatively` in this sentence.

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md [hunk 5]
```diff
@@ -493,34 +502,92 @@ Details about these fields are as follows:
...
 
-The `insecureSkipVerify` field is used to determine whether the client will verify the server's certificate chain and hostname. The default value is `false`. In production environments, it is recommended to set this to `false` to ensure the security of communication. If set to `true`, the client will not verify the server's certificate chain and hostname.
+  The Nebula-Operator chart allows configuration of the init container, sidecar container, volumes, and volumeMounts within the controller-manager Pod. It accesses the certificates generated by the sidecar-container through the environment variables `CA_CERT_PATH`, `CLIENT_CERT_PATH`, and `CLIENT_KEY_PATH`. To generate certificates, you need to place the sidecar-container and the controller-manager container in the same Pod and share the certificate directory credentials. The controller-manager then reads the client-side certificates from the environment variables and establishes a mutual TLS (mTLS) connection with the Meta and Storage services. Example:
```
Comment (wenhaocs) [inline docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md:513]: Please specify the following configurations are for operator.

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md [hunk 6]
```diff
@@ -493,34 +502,92 @@ Details about these fields are as follows:
 
 ### Configure `sslCerts`
 
-When you enable mTLS between services, you still needs to set `spec.sslCerts`, because NebulaGraph Operator communicates with the Meta service and Storage service. 
+As NebulaGraph Operator needs to communicate with the Meta and Storage services, you must configure `spec.sslCerts` to set certificates for it when enabling mTLS between services.
```
Comment (wenhaocs) [inline docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md:505]: Please state clearly which yaml `spec.sslCerts` is in.

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md [hunk 7]
```diff
@@ -493,34 +502,92 @@ Details about these fields are as follows:
...
-```yaml
-spec:
-  sslCerts:
-    clientSecret: "client-cert" 
-    clientCert: "tls.crt"  # The key name of the certificate in the client certificate Secret, default is "tls.crt".
-    clientKey: "tls.key"   # The key name of the private key in the client certificate Secret, default is "tls.key".
-    caSecret: "ca-cert" # The name of the CA certificate Secret.
-    caCert: "tls.crt"  # The key name of the certificate in the CA certificate Secret, default is "tls.crt".
-```
+You can create a Secret to pass in the certificates or configure the environment variables of NebulaGraph Operator to obtain local certificates.
```
Comment (wenhaocs) [inline docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md:509]: Please make it clear that when environment variable are used, the secrets under the `spec.sslCerts` section will be ignored.

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md (hunk 5 above)
Comment (wenhaocs) [reply docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md:513]: And state that once environment variables are set in **operator**, all the secrets in `sslCerts` in the cluster configuration will be ignored.

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md (hunk 4 above)
Comment (abby-cyber) [reply docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md:557]: Should the following configurations be an alternative for Operator?

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md (hunk 6 above)
Comment (abby-cyber) [reply docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md]: done

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md (hunk 5 above)
Comment (abby-cyber) [reply docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md]: done

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md (hunk 3 above)
Comment (abby-cyber) [reply docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md]: done

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md (hunk 7 above)
Comment (abby-cyber) [reply docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md]: done

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md (hunk 4 above)
Comment (wenhaocs) [reply docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md]: Yes, it's "either..or" between using secrets or environmental variable.

Comment (wenhaocs) [reply docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md]: Please make it clear that it's `either...or` from line 513 to line 557.

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md (hunk 1 above)
Comment (wenhaocs) [reply docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md]: Please set it to true

File: docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md (hunk 4 above)
Comment (wenhaocs) [reply docs-2.0/k8s-operator/4.cluster-administration/4.7.security/4.7.1.enable-mtls.md]: Synced offline. Can resolve this.

**Pull request merged**